openedx/features/pakx/lms/pakx_admin_app/views.py

```diff
@@ -436,3 +438,26 @@ def get(self, *args, **kwargs):  # pylint: disable=unused-argument
             user_info['role'] = TRAINING_MANAGER if user_groups.name == GROUP_TRAINING_MANAGERS else ORG_ADMIN
 
         return Response(status=status.HTTP_200_OK, data=user_info)
+
+
+class CourseListAPI(generics.ListAPIView):
+    authentication_classes = [SessionAuthentication]
+    permission_classes = [CanAccessPakXAdminPanel]
+    pagination_class = PakxAdminAppPagination
+    serializer_class = CoursesSerializer
+
+    PakxAdminAppPagination.page_size = 3
+
+    instructors = {}
+
+    def get_serializer_context(self):
+        context = super(CourseListAPI, self).get_serializer_context()
+        context.update({"instructors": self.instructors})
+        return context
+
+    def get_queryset(self):
+        queryset = CourseOverview.objects.all()
+        course_access_role_qs = CourseAccessRole.objects.filter(course_id__in=queryset.values_list('id'))
+        for course_access_role in course_access_role_qs:
+            self.instructors[course_access_role.course_id] = course_access_role.user.username
```

Comment:
what if a course has multiple instructors? keep a list of users & show their name instead of username